sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -315,12 +366,7 @@ public void close() {
     private Mono<Void> sendDispositionImpl(String deliveryTag, DeliveryState desiredState) {
         final Delivery delivery = deliveries.get(deliveryTag);
         if (delivery == null) {
-            logger.atWarning()
-                .addKeyValue(DELIVERY_TAG_KEY, deliveryTag)
-                .log("Delivery not found to update disposition.");
-
-            return monoError(logger, Exceptions.propagate(new IllegalArgumentException(
-                "Delivery not on receive link.")));
+            return monoError(logger, DeliveryNotOnLinkException.noMatchingDelivery(deliveryTag, desiredState));
```

Comment:
line 423:  Can we remove the linkName and deliveryTag as kvps and remove it from teh log line?

Reply:
line 504: consider adding call site and locks as part of kvp? (or maybe we can log multiple items in a key). I think it'll be easier to read the logs.

Reply:
There are a few places where we still use `linkName[%s]` but I don't want to clog this discussion up. :)

Reply:
Yep, let's move these to log key-values.  Ops, there was also an incorrect positional argument, will fix that also,